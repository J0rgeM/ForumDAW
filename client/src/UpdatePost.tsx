import React, { useEffect, useState } from "react";
// Library imports.
import axios, { AxiosResponse } from "axios";
import Navbar from "./components/Navbar";
import Footer from "./components/Footer";

import { Simulate } from "react-dom/test-utils";
import error = Simulate.error;
import { IPost } from "../../server/src/posts";
import { useParams } from "react-router-dom";

function UpdatePost() {
    let params = useParams();
    const [post, setPost] = useState<IPost[]>([]);
    const [author, setAuthor] = useState("");
    const [body, setBody] = useState("");

    const getPost = async (nmb: any) => {
        try {
            const response: AxiosResponse = await axios.get<IPost[]>(
                "http://localhost:8080/updatePost/" + String(nmb)
            );
            setPost(response.data);
            console.log(response.data);
        } catch (error) {
            console.error(error);
        }
    };

    // eslint-disable-next-line
    const updatePost = async (e, nmb: any) => {
        e.preventDefault();
        try {
            const response: AxiosResponse = await axios.put(
                "http://localhost:8080/updatePost/" + String(nmb),
                {
                    nmb: nmb,
                    author: JSON.stringify(author),
                    body: JSON.stringify(body),
                    headers: {
                        "Content-type": "text/plain",
                    },
                }
            );
            console.log(response);
        } catch (error) {
            console.error(error);
        }
        window.location.href = "http://localhost:3000";
    };
    useEffect(() => {
        getPost(params.number);
    }, []);

    return (
        <div>
            <Navbar />
            {post.map((post) => (
                <div key={post.nmb} className="container text-center">
                    <form
                        onSubmit={(e) => updatePost(e, post.nmb)}
                        className="shadow p-3 mb-5 bg-white rounded"
                    >
                        <h1 className="mt-2">Post</h1>
                        <div className="mb-3">
                            <input
                                type="textarea"
                                className="form-control m-1"
                                id="inputPost"
                                onChange={(e) => setAuthor(e.target.value)}
                                placeholder={post.author}
                            ></input>
                            <input
                                type="textarea"
                                className="form-control m-1"
                                id="inputPost"
                                onChange={(e) => setBody(e.target.value)}
                                placeholder={post.body}
                            ></input>
                        </div>
                        <button type="submit" className="btn btn-dark">
                            Submit
                        </button>
                    </form>
                </div>
            ))}

            <Footer />
        </div>
    );
}

export default UpdatePost;
