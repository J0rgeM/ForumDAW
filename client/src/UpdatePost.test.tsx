import {describe, expect, it} from '@jest/globals';
import {renderWithRouter} from "./lib";
import UpdatePost from "./UpdatePost";

describe('UpdatePost', () => {
    it('match snapshot', () => {
        const { container} = renderWithRouter(<UpdatePost/>)
        expect(container).toMatchSnapshot()
    })
})