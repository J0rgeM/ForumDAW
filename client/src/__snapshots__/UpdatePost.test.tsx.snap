// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatePost match snapshot 1`] = `
<div>
  <div>
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
           BlogGIF
        </a>
        <button
          aria-controls="navbarNavDropdown"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarNavDropdown"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse justify-content-end"
          id="navbarNavDropdown"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="nav-link active rounded-3"
                href="/addPost"
              >
                Add Post
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="nav-link active rounded-3"
                href="/"
              >
                List Posts
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      class="footer text-center fixed-bottom text-white bg-dark"
    >
      <div
        class="container-fluid"
      >
        <div
          class="row"
        >
          <p
            class="text-center"
          >
            Â© 2023 DAW
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
